samples/objc/FirebaseUI-demo-objc/Samples/Auth/FUIAuthViewController.m

```diff
@@ -255,14 +285,25 @@ - (NSString *)getAllIdTokens {
 
 - (void)signOut {
   NSError *error;
-  [self.authUI signOutWithError:&error];
-  if (error) {
-    [self showAlert:error.localizedDescription];
+  FIRUser *currentUser = self.authUI.auth.currentUser;
+  // If the user is anonymous, delete the user to avoid dangling anonymous users.
+  if (currentUser.isAnonymous) {
+    [currentUser deleteWithCompletion:^(NSError * _Nullable error) {
+      if (error) {
+        [self showAlertWithTitlte:@"Error" message:error.localizedDescription];
+      }
+      [self showAlertWithTitlte:@"" message:@"Anonymous user deleted"];
```

Comment:
In case of error 2 alerts are shown. Do we need to display 2nd alert?

Reply:
After the first alert there should be a return statement, added.